### PR TITLE
Support topN for Snowflake connector

### PR DESCRIPTION
**Pull request by chenjian2664**
## Release notes

(x) Release notes are required, with the following suggested text:

```markdown
# Snowflake
* Add support for top-N pushdown. ({issue}`21219`)
```

### CODE REVIEW

File: plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java [hunk 1]
```diff
@@ -283,6 +284,31 @@ public boolean isLimitGuaranteed(ConnectorSession session)
...
 
+    @Override
+    public boolean supportsTopN(ConnectorSession session, JdbcTableHandle handle, List<JdbcSortItem> sortOrder)
+    {
+        for (JdbcSortItem sortItem : sortOrder) {
+            Type sortItemType = sortItem.getColumn().getColumnType();
+            if (sortItemType instanceof CharType || sortItemType instanceof VarcharType) {
+                // Remote database can be case insensitive.
+                return false;
+            }
```
Comment (ebyhr) [inline plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java:295]: Is it true? or did you just copy from other connectors?

**Comment by ebyhr** [comment]
Please add the following line to snowflake.md:
```
- {ref}`topn-pushdown`
```

File: plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java (hunk 1 above)
Comment (chenjian2664) [reply plugin/trino-snowflake/src/main/java/io/trino/plugin/snowflake/SnowflakeClient.java:295]: see https://docs.snowflake.com/en/sql-reference/collation#supported-specifiers

**Comment by ebyhr** [comment]
/test-with-secrets sha=ef2e7687fd39a271625087fef8dfafa0045a93b0

**Pull request merged**